docs/hyperparameter_search.rst

```diff
@@ -0,0 +1,136 @@
+.. inclusion-marker-start-do-not-remove
+Distributed Hyperparameter Search
+=================================
+
+Horovod's data parallelism training capabilities allows you to scale out and speed up the workload of training a deep learning model. However, distributed training often does not exhibit linear scaling in accuracy improvement - i.e., simply using 2x more workers does not necessarily mean the model will obtain the same accuracy in 2x less time.
+
+To address this, you will often need to re-tune hyperparameters when training at scale, as many hyperparameters exhibit different behaviors at larger scales.
+
+To address this issue, Horovod offers a Ray Tune integration to enable parallel hyperparameter tuning with distributed training.
+
+Ray Tune is an industry standard tool for distributed hyperparameter tuning. Ray Tune includes the latest hyperparameter search algorithms, integrates with TensorBoard and other analysis libraries, and natively supports distributed training. The Ray Tune + Horovod integration leverages the underlying Ray framework to provide a scalable and comprehensive hyperparameter tuning setup.
+
+**By the end of this guide, you will learn:**
+* How you can set up Ray Tune and Horovod to tune your hyperparameters
+* Typical hyperparameters to configure for distributed training
+
+Horovod + Ray Tune
+------------------
+
+You can leverage Ray Tune with Horovod to combine distributed hyperparameter tuning with distributed training. Here is an example demonstrating basic usage:
+
+.. code-block:: python
+
+    import horovod.torch as hvd
+    from ray import tune
+    import time
+
+    def training_function(config: Dict):
+        hvd.init()
+        for i in range(config["epochs"]):
+            time.sleep(1)
+            tune.report(test=1, rank=hvd.rank())
+
+    trainable = DistributedTrainableCreator(
+            training_function, num_slots=2, use_gpu=use_gpu)
+    analysis = tune.run(
+            trainable, num_samples=2, config={"epochs": tune.grid_search([1, 2, 3]))
+    print(analysis.best_config)
+
+Not only is the interface very simple, it is also packed with features.
+
+Basic setup
+-----------
+
+Use the `DistributedTrainableCreator`_ function to adapt your Horovod training function to be compatible with Tune.
+
+`DistributedTrainableCreator`_ exposes ``num_hosts``, ``num_slots``, ``use_gpu`` and ``num_cpus_per_slot`` - these parameters allow you to specify the resource allocation of a single "trial" (or "Trainable") which itself can be a distributed training job.
```

Comment:
"- these parameters allow you to specify" --> ". Use these parameters to specify"